### ISSUE TITLE
Segfault when clicked on the advanced tab

### ISSUE THREAD

**Issue by arpadlukacs** [privacore/open-source-search-engine#45]
I recompiled the stable version of the engine in debug mode. Here is the content of the lastcore.bt.txt

```
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `./gb -l'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000561d4f in PosdbTable::delNonMatchingDocIdsFromSubLists (this=0x7f2077fcdfd8) at PosdbTable.cpp:4936
4936                            float maxMatchingTermFreqWeight = qti->m_subList[qti->m_matchingSublist[0].m_baseSubListIndex].
#0  0x0000000000561d4f in PosdbTable::delNonMatchingDocIdsFromSubLists (this=0x7f2077fcdfd8) at PosdbTable.cpp:4936
#1  0x000000000055b0d3 in PosdbTable::findCandidateDocIds (this=0x7f2077fcdfd8) at PosdbTable.cpp:2137
#2  0x000000000055eb24 in PosdbTable::intersectLists_real (this=0x7f2077fcdfd8) at PosdbTable.cpp:3675
#3  0x000000000055e84f in PosdbTable::intersectLists (this=0x7f2077fcdfd8) at PosdbTable.cpp:3620
#4  0x000000000048472a in Msg39::intersectListsThreadFunction (state=0x7f2078a3dbc0) at Msg39.cpp:843
#5  0x000000000045e298 in (anonymous namespace)::job_pool_thread_function (pv=0x7f20740030d0) at JobScheduler.cpp:138
#6  0x00007f208c52d6ba in start_thread (arg=0x7f2072ffd700) at pthread_create.c:333
#7  0x00007f2089d5f41d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109

```

Interestingly, it does not always crash when we go to the advanced tab. Eg I restarted it and went straight there and now it does not crash.

Is it ok if I keep posting any crash we experience? For now, it seems like it crashes randomly, but maybe you can help us with these.

Thank you

**Comment by br-privacore** [comment]
The stable branch is pretty old. I'd suggest you try the newest from master.

**Comment by arpadlukacs** [comment]
Ok, I rebuild the master and it crashed right after I tried to start:

```
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".                                                     
Core was generated by `./gb -l'.                                                                                               
Program terminated with signal SIGABRT, Aborted.                                                                               
#0  0x00007fdcdf9dc428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54                                
54      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.                                                         
#0  0x00007fdcdf9dc428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54                                
#1  0x00007fdcdf9de02a in __GI_abort () at abort.c:89                                                                          
#2  0x00000000004eab50 in Process::shutdown2 (this=0x11ec7e0 <g_process>) at Process.cpp:752                                   
#3  0x00000000004ea5de in Process::shutdown (this=0x11ec7e0 <g_process>, urgent=true, state=0x0, callback=0x0) at Process.cpp:5
#4  0x00000000004ea515 in Process::shutdownAbort (this=0x11ec7e0 <g_process>, save_on_abort=true) at Process.cpp:502           
#5  0x000000000055c74f in Msg3::reset (this=0x7fdc3d329ca0) at Msg3.cpp:61                                                     
#6  0x0000000000560a93 in Msg5::reset (this=0x7fdc3d329c90) at Msg5.cpp:80
#7  0x00000000005609fd in Msg5::~Msg5 (this=0x7fdc3d329c90, __in_chrg=<optimized out>) at Msg5.cpp:67
#8  0x00000000004779ec in Msg0::reset (this=0x7fdcbe9763f8) at Msg0.cpp:68
#9  0x0000000000477986 in Msg0::~Msg0 (this=0x7fdcbe9763f8, __in_chrg=<optimized out>) at Msg0.cpp:62
#10 0x00000000005194e8 in SiteGetter::~SiteGetter (this=0x7fdcbe976320, __in_chrg=<optimized out>) at SiteGetter.cpp:92
#11 0x0000000000600bae in XmlDoc::getNewTagBuf (this=0x7fdc3d330440) at XmlDoc.cpp:19586
#12 0x00000000005ef96a in XmlDoc::getMetaList (this=0x7fdc3d330440, forDelete=false) at XmlDoc.cpp:13196
#13 0x00000000005d237f in XmlDoc::indexDoc2 (this=0x7fdc3d330440) at XmlDoc.cpp:2046
#14 0x00000000005d136d in XmlDoc::indexDoc (this=0x7fdc3d330440) at XmlDoc.cpp:1673
#15 0x00000000005d07a6 in indexDoc3 (state=0x7fdc3d330440) at XmlDoc.cpp:1328
#16 0x00000000005d08ff in indexDocWrapper (state=0x7fdc3d330440) at XmlDoc.cpp:1373
#17 0x0000000000497186 in Msge1::gotMsgCWrapper (state=0x7fdc3d351b10, ip=370858169) at Msge1.cpp:285
#18 0x000000000048356a in gotReplyWrapper (state=0x7fdc3d351b10, state2=0x7fdc3d351b10) at MsgC.cpp:198
#19 0x0000000000499675 in Multicast::closeUpShop (this=0x7fdc3d351b28, slot=0x7fdc45c5428c) at Multicast.cpp:979
#20 0x00000000004994fb in Multicast::gotReply1 (this=0x7fdc3d351b28, slot=0x7fdc45c5428c) at Multicast.cpp:927
#21 0x0000000000498f4f in Multicast::gotReply1 (state=0x7fdc3d351b28, slot=0x7fdc45c5428c) at Multicast.cpp:797
#22 0x00000000005c2370 in UdpServer::makeCallback (this=0x128c000 <g_udpServer>, slot=0x7fdc45c5428c) at UdpServer.cpp:1528
#23 0x00000000005c1e2a in UdpServer::makeCallbacks (this=0x128c000 <g_udpServer>, niceness=1) at UdpServer.cpp:1377
#24 0x00000000005c0d2f in UdpServer::process (this=0x128c000 <g_udpServer>, now=1517686256533, maxNiceness=100) at UdpServer.cpp:865
#25 0x00000000005c0dbd in UdpServer::readPollWrapper (fd=22, state=0x128c000 <g_udpServer>) at UdpServer.cpp:879
#26 0x00000000005552f4 in Loop::callCallbacks_ass (this=0x127a4e0 <g_loop>, forReading=true, fd=22, now=1517686256533, niceness=0) at Loop.cpp:421
#27 0x0000000000556268 in Loop::doPoll (this=0x127a4e0 <g_loop>) at Loop.cpp:788
#28 0x0000000000555b6c in Loop::runLoop (this=0x127a4e0 <g_loop>) at Loop.cpp:662
#29 0x00000000004183b9 in main2 (argc=2, argv=0x7ffe3f8f31e8) at main.cpp:1633
#30 0x0000000000413e33 in main2_trampoline () at main.cpp:219
#31 0x00007fdce26936ba in start_thread (arg=0x7fdcbe981700) at pthread_create.c:333
#32 0x00007fdcdfaae41d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109

```

**Comment by alc-privacore** [comment]
Fixed on master commit 773c5f1e9327066a26b12887ceb321b3979da84a
